experimental/Origami/src/types.jl

```diff
@@ -0,0 +1,6 @@
+mutable struct Origami
+    o::GapObj
+    h::PermGroupElem
+    v::PermGroupElem
+    d::Oscar.Integer
```

Comment:
Don't use abstract types like `Oscar.Integer` in a struct if you can avoid it -- and here you certainly can: I think `d` is a "degree" and even fits into a "small" machine integer:
```suggestion
    d::Int
```
But perhaps my guess is wrong and this is a value that can be arbitrarily large. Then I would recommend using a `ZZRingElem` (the OSCAR type for big integers, similar to but different from Julia's own `BigInt` type)